source/common/ssl/context_impl.cc

```diff
@@ -200,7 +200,7 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
         ASN1_STRING* str = altname->d.dNSName;
         char* dns_name = reinterpret_cast<char*>(ASN1_STRING_data(str));
         for (auto& config_san : subject_alt_names) {
-          if (dNSNameMatch(config_san, dns_name)) {
+          if (dNSNameMatch(config_san, dns_name, static_cast<size_t>(ASN1_STRING_length(str)))) {
```

Comment:
To avoid extra calls to `ASN1_STRING_length()` (since I don't think it will get inlined) and dereference of `str`.